### DISCUSSION TITLE
IR & Deep Sleep Mode

### DISCUSSION THREAD

**Discussion by gwinterhalter** [Arduino-IRremote/Arduino-IRremote#1142] [General]
Hi All:

I am trying to use this library to read IR Sensor with ESP32 Deep Sleep Mode but the define function is not recognizing any remote streams. Everything works fine when using active mode. Sample sketch would be welcome or advice.

Thanks

Greg

**Comment by ArminJo** [comment]
How long does it take to wake up from the deep sleep?

**Reply by gwinterhalter** [reply]
around 170ms. Using Simple IR) Receiver sketch, I am able to get the remote codes. When I reformat sketch into deep sleep format, it no longer can receive. very perplexing. I have also create a loop to check for new message with irreive.define but it returns 0 always.

**Reply by ArminJo** [reply]
You are familiar with the average IR timing? Especially the header timing?

**Comment by AnalysIR** [comment]
Design your code so that the first IR signal wakes the system up for 1->5 secs and then if you press the remote twice the second signal can be captured.
Note on some protocols just keeping the button pressed will not work (eg NEC....repeat signal is different from the main signal).

If you are sending the signal from a micro...just send the signal twice with a 500ms gap between each.

You only need to turn on WiFi etc if using it...so you can keep the power usage low.
